runtime/minor_gc.c

```diff
@@ -464,7 +464,7 @@ CAMLexport void caml_gc_dispatch (void)
 }
 
 /* Called by [Alloc_small] when [caml_young_ptr] reaches [caml_young_limit].
-   We have to either call memprof or the gc. */
+   We may have to either call memprof or the gc. */
```

Comment:
If a finaliser (or other async callback) is pending, then in native code this function will be called on each minor allocation even when the heap is not exhausted.  The code of the function will correctly do nothing in this case, but the comment suggested otherwise.